src/lib/hosts-file.ts

```diff
@@ -0,0 +1,181 @@
+import fs from 'fs';
+import { domainToASCII } from 'node:url';
+import { platform, tmpdir } from 'os';
+import path from 'path';
+import { promisify } from 'util';
+import * as Sentry from '@sentry/electron/main';
+import sudo from 'sudo-prompt';
+
+const readFile = promisify( fs.readFile );
+const writeFile = promisify( fs.writeFile );
+const sudoExec = promisify( sudo.exec );
+
+// Host file paths for different operating systems
+const HOST_FILES: Record< string, string > = {
+	win32: path.resolve( process.env.SystemRoot ?? 'C:\\Windows', 'System32\\drivers\\etc\\hosts' ),
+	darwin: '/etc/hosts',
+	linux: '/etc/hosts',
+};
+
+// Get the appropriate hosts file path for the current platform
+const getHostsFilePath = (): string => {
+	const currentPlatform = platform();
+	const hostsPath = HOST_FILES[ currentPlatform ];
+
+	if ( ! hostsPath ) {
+		throw new Error( `Unsupported platform: ${ currentPlatform }` );
+	}
+
+	return hostsPath;
+};
+
+/**
+ * Reads the hosts file content
+ */
+export const readHostsFile = async (): Promise< string > => {
+	try {
+		const hostsPath = getHostsFilePath();
+		const content = await readFile( hostsPath, 'utf8' );
+		return content;
+	} catch ( error ) {
+		console.error( 'Error reading hosts file:', error );
+		throw error;
+	}
+};
+
+/**
+ * Writes content to the hosts file (requires elevated permissions)
+ */
+export const writeHostsFile = async ( content: string ): Promise< void > => {
+	const hostsPath = getHostsFilePath();
+	try {
+		const tempPath = path.join( tmpdir(), 'wp-studio-hosts' );
+		await writeFile( tempPath, content );
+		const command =
+			platform() === 'win32'
+				? `type ${ tempPath } > ${ hostsPath }`
+				: `cat ${ tempPath } > ${ hostsPath }`;
+		// @ts-expect-error promisify doesn't seem typed properly.
+		await sudoExec( command, {
+			name: 'WordPress Studio',
+		} );
+	} catch ( error ) {
+		console.error( 'Error writing hosts file:', error );
+		throw error;
+	}
+};
+
+/**
+ * Create a regular expression matching the hosts entry for a given domain:
+ *
+ * 	127.0.0.1 foo.wp.cloud # Port 8000 (WordPress Studio)
```

Comment:
```suggestion
 * 	127.0.0.1 foo.wp.cloud # Port 8000
```
Nit, but let's adjust this now that we no longer add `(WordPress Studio)` to each entry